pkg/virt-controller/watch/vmi.go

```diff
@@ -778,6 +791,29 @@ func (c *VMIController) updateStatus(vmi *virtv1.VirtualMachineInstance, pod *k8
 	return nil
 }
 
+func isVirtHandlerAlive(clientset kubecli.KubevirtClient, nodeName string) (bool, error) {
+	handlerNodeSelector := fields.ParseSelectorOrDie("spec.nodeName=" + nodeName)
+	handlerLabelSelector := labels.Set{virtv1.AppLabel: "virt-handler"}.AsSelector()
+	list, err := clientset.CoreV1().Pods(v1.NamespaceAll).List(context.Background(), v1.ListOptions{
+		FieldSelector: handlerNodeSelector.String(),
+		LabelSelector: handlerLabelSelector.String(),
+	})
+
+	if err != nil {
+		return false, fmt.Errorf("failed to get virt-handler pods %v", err)
+	}
```

Comment:
A failure to get virt-handler pods should not result in virt-controller marking vmi's as failed. that's an api failure that doesn't reflect whether there's a virt-handler online or not.

Reply:
Done.